src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -214,17 +228,37 @@ def add_filesystem(self, spec):
             }
         }
 
-        try:
-            self.rook_api_post(
-                "cephfilesystems/",
-                body=rook_fs
-            )
-        except ApiException as e:
-            if e.status == 409:
-                log.info("CephFilesystem '{0}' already exists".format(spec.name))
-                # Idempotent, succeed.
-            else:
-                raise
+        with self.ignore_409("CephFilesystem '{0}' already exists".format(spec.name)):
+            self.rook_api_post("cephfilesystems/", body=rook_fs)
+
+    def add_nfsgw(self, spec):
+        # TODO use spec.placement
+        # TODO use spec.min_size (and use max_size meaningfully)
+        # TODO warn if spec.extended has entries we don't kow how
+        #      to action.
+
+        rook_nfsgw = {
+            "apiVersion": ROOK_API_NAME,
+            "kind": "CephNFS",
+            "metadata": {
+                "name": spec.name,
+                "namespace": self.rook_namespace
+            },
+            "spec": {
+                "RADOS": {
+                    "pool": spec.extended["pool"]
+                },
+                "server": {
+                    "active": spec.max_size,
```

Comment:
This would be the active count for the given NFS cluster?

Reply:
Perhaps a better name would be NFSGWCluster if the active count can be > 1.

Reply:
Yes, that's the number of servers in the given cluster. We can suggest a new name to @travisn but he may get fed up if he has to change it again. ;)